cli/internal/ffi/ffi.go

```diff
@@ -0,0 +1,75 @@
+package ffi
+
+// #include "bindings.h"
+//
+// #cgo LDFLAGS: -L${SRCDIR} -lturborepo_ffi
+// #cgo windows LDFLAGS: -lole32 -lbcrypt -lws2_32 -luserenv
+import "C"
+
+import (
+	"reflect"
+	"unsafe"
+
+	ffi_proto "github.com/vercel/turbo/cli/internal/ffi/proto"
+	"google.golang.org/protobuf/proto"
+)
+
+// Unmarshal consumes a buffer and parses it into a proto.Message
+func Unmarshal[M proto.Message](b C.Buffer, c M) error {
```

Comment:
I _think_ this is our first generic into the Go codebase. I don't think it matters anymore, but I want to make sure we didn't just break linting.

/cc @gsoltis